grails-app/services/org/pih/warehouse/inventory/CycleCountService.groovy

```diff
@@ -390,33 +374,31 @@ class CycleCountService {
         return cycleCounts.collect { CycleCountDto.toDto(it) }
     }
 
+    /**
+     * Submits the (re)count as it is in its current state.
+     */
     CycleCountDto submitCount(CycleCountSubmitCountCommand command) {
-        command.cycleCount.cycleCountItems.each { CycleCountItem cycleCountItem ->
-            updateCycleCountItemForSubmit(cycleCountItem, command.refreshQuantityOnHand, command.failOnOutdatedQuantity)
+        CycleCount cycleCount = command.cycleCount
+
+        if (command.refreshQuantityOnHand) {
+            boolean hasQuantityChanged = cycleCountProductAvailabilityService.refreshProductAvailability(cycleCount)
+            if (hasQuantityChanged && command.failOnOutdatedQuantity) {
```

Comment:
Right now, the failOn is probably up to the frontend developer but it would be good to wire this into a feature flag that can be used to set this value on the frontend. i.e. instead of hard-coding this in React, we just pull from a the feature flag from the AppContext API.